app/src/main/java/org/ole/planet/myplanet/services/DownloadService.kt

```
@@ -486,14 +486,12 @@ class DownloadService : Service() {
-    private fun onDownloadComplete(url: String) {
+    private suspend fun onDownloadComplete(url: String) {
         if ((outputFile?.length() ?: 0) > 0) {
-            appScope.launch {
-                try {
-                    resourcesRepository.markResourceOfflineByUrl(url)
-                } catch (e: Exception) {
-                    e.printStackTrace()
-                }
+            try {
+                resourcesRepository.markResourceOfflineByUrl(url)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)